### PR TITLE
Add dfe-analytics

### PR DESCRIPTION
**Pull request by CatalinVoineag**
## Context

This will send data on every web request and whenever a table in config/analytics is updated/created. Regardless of domain. We don't have a way to distinguish between them at the moment, though the web requests do contain the domain.

A scheduled job is also going to run every day after midnight to keep in sync the BigQuery data with our own data.

We will send data to dfe-analytics in Production, Sandbox, Staging and QA.

I have included every table that we have apart from the ones that our gems generate. Some contain personal information, they have been included in `config/analytics_pii.yml` to handle the sensitive information. 

## Changes proposed in this pull request

Dfe analytics setup

## Guidance to review

Review code.
This will only work when the envs have the secret keys

## Screenshots

https://github.com/DFE-Digital/itt-mentor-services/assets/11318084/afbe1544-0ed2-452b-8e87-107b15785ee2

### CODE REVIEW

File: spec/services/concerns/service_pattern_spec.rb [hunk 1]
```diff
@@ -1,4 +1,5 @@
 require "rails_helper"
+require "./app/services/concerns/service_pattern" # don't get this from dfe-analytics
```
Comment (CatalinVoineag) [inline spec/services/concerns/service_pattern_spec.rb:2]: The spec will include the service pattern from the dfe analytics gem without this 🙃

**Comment by CatalinVoineag** [comment]
[Set up data pipeline (DfE analytics)](https://trello.com/c/u9W4PKnG/17-set-up-data-pipeline-dfe-analytics)

File: spec/services/concerns/service_pattern_spec.rb (hunk 1 above)
Comment (Nitemaeric) [reply spec/services/concerns/service_pattern_spec.rb:2]: 🙃

Comment (ollietreend) [reply spec/services/concerns/service_pattern_spec.rb:2]: @CatalinVoineag @Nitemaeric FYI I've reported the issue so the dfe-analytics team are aware: https://github.com/DFE-Digital/dfe-analytics/issues/136

It's on their radar now. But we'll need this workaround for the time being.

Perhaps it'd be useful to add a comment to this line to reference that GitHub issue?

```suggestion
require "./app/services/concerns/service_pattern" # needed until DFE-Digital/dfe-analytics#136 is fixed
```

**Review by ollietreend — APPROVED** [review]
This looks good to me. Thanks for setting it up @CatalinVoineag ⭐

But before merging this PR, could you please add a reference to the dfe-analytics issue in the code comment? As per https://github.com/DFE-Digital/itt-mentor-services/pull/521#discussion_r1579296024

Thanks 🙇🏻

**Review by Nitemaeric — APPROVED** [review]

**Pull request merged**